spec/rubocop/cop/betterment/authorization_in_controller_spec.rb

```diff
@@ -2,225 +2,190 @@
 
 require 'spec_helper'
 
-describe RuboCop::Cop::Betterment::AuthorizationInController, :config do
-  let(:offense_create) do
-    <<~MSG
-      Model created/updated using unsafe parameters.
-      Please query for the associated record in a way that enforces authorization (e.g. "trust-root chaining"),
-      and then pass the resulting object into your model instead of the unsafe parameter.
-
-      INSTEAD OF THIS:
-      post_parameters = params.permit(:album_id, :caption)
-      Post.new(post_parameters)
-
-      DO THIS:
-      album = current_user.albums.find(params[:album_id])
-      post_parameters = params.permit(:caption).merge(album: album)
-      Post.new(post_parameters)
-
-      See here for more information on this error:
-      https://github.com/Betterment/betterlint/blob/main/README.md#bettermentauthorizationincontroller
-    MSG
-  end
-
+describe RuboCop::Cop::Betterment::AuthorizationInController, :betterlint_config do
   context 'when creating or updating a model' do
     it 'registers an offense for unsafe parameters' do
-      inspect_source(<<~RUBY)
+      expect_offense(<<~RUBY)
         class Application
           def create
             Model.new params[:user_id]
+                      ^^^^^^^^^^^^^^^^ Model created/updated [...]
             Model.create! params[:user_id]
+                          ^^^^^^^^^^^^^^^^ Model created/updated [...]
           end
         end
       RUBY
-
-      expect(cop.offenses.size).to be(2)
-      expect(cop.offenses.map(&:line)).to eq([3, 4])
-      expect(cop.highlights.uniq).to eq(['params[:user_id]'])
-      expect(cop.messages.uniq).to eq([offense_create])
     end
```

Comment:
Referring `cop.offenses` raises an error now. Instead, you can use the `offenses` returned from `inspect_source`, but there's no way to get the `highlights`.

Offenses do have a `#highlighted_area` method, but [it appears to be broken](https://github.com/rubocop/rubocop/issues/12527).

`expect_offense` is the correct way to test cops. [`inspect_source` is likely to be removed in the future](https://github.com/rubocop/rubocop/issues/8003).